app/src/main/java/com/youssef/fixit/Auth/LoginFragment/LoginFragment.java

```diff
@@ -159,12 +158,17 @@ public void OnForgotPassword() {
         Toast.makeText(getContext(), "Mail is Sent", Toast.LENGTH_SHORT).show();
     }
 
+    @Override
+    public void ShowLoading() {
+        dialog.show();
+    }
+
     void SaveDataInShared(String token, String role, int my_id) {
         editor.putString("token", token);
         editor.putString("role", role);
         editor.putInt("my_id", my_id);
         editor.commit();
-        getActivity().startActivity(new Intent(getContext(), MainActivity.class));
+        getActivity().startActivity(new Intent(getContext(), HomeActivity.class));
```

Comment:
For the future, we can discuss when to use fragment and activity to avoid this needless stacks of fragments and activity

For now, if the activity doesn't have multiple fragments, better to keep it one activity and no fragment attached.
For example, AuthActivity has only one fragment and there's no future for any added fragment, then `AuthActivity` is enough to handle the view no need to have this `loginFragment`

What do you think?